backend/de.metas.monitoring/src/main/java/de/metas/monitoring/adapter/PerformanceMonitoringService.java

```diff
@@ -58,54 +58,28 @@ default void monitorSpan(final Runnable runnable, final SpanMetadata metadata)
 	
 	@Value
 	@Builder
-	class SpanMetadata
+	class Metadata
 	{
 		@NonNull
 		String name;
 
 		@NonNull
-		String type;
+		Type type;
 
 		@Nullable
-		String subType;
+		SubType subType;
 
 		@Nullable
 		String action;
 
-		@Singular
-		Map<String, String> labels;
-
 		@Nullable
-		BiConsumer<String, String> distributedHeadersInjector;
-	}
-
-	/** Invoke the given {@code callable} as a transaction. Capture exception and re-throw it, wrapped as RuntimeException if required. */
-	<V> V monitorTransaction(Callable<V> callable, TransactionMetadata request);
-
-	default void monitorTransaction(
-			@NonNull final Runnable runnable,
-			@NonNull final TransactionMetadata request)
-	{
-		monitorTransaction(
-				() -> {
-					runnable.run();
-					return null;
-				},
-				request);
-	}
-
-	@Value
-	@Builder
-	class TransactionMetadata
-	{
-		@NonNull
-		String name;
+		String window;
```

Comment:
is this the "windowInfo"?

Reply:
yes, window name + id (changed to windowNameAndId for better understanding)